From 6ad21f48855b39fee4cc92cacf7d9441f62737a0 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:38 -0400
Subject: [usb] xhci: Change compatible string from xhci-platform to generic-xhci

Message-id: <1410887463-20674-267-git-send-email-dzickus@redhat.com>
Patchwork-id: 95355
O-Subject: [RHEL7 PATCH 266/591] xhci-platform: Change compatible string from xhci-platform to generic-xhci
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 0f94388b27c599015b74eedf1a32126a3f5fc0f9
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Tue Feb 11 17:54:46 2014 +0100

    xhci-platform: Change compatible string from xhci-platform to generic-xhci

    This brings the xhci-platform bindings in sync with what we've done for
    the ohci- and ehci-platform drivers. As discussed there using platform as a
    postfix is a bit weird as the platform bus is a Linux specific thing and
    the bindings are supposed to be OS agnostic.

    Note that the old xhci-platform compatible string is kept around for, well,
    compatibility reasons.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/Documentation/devicetree/bindings/usb/usb-xhci.txt b/Documentation/devicetree/bindings/usb/usb-xhci.txt
index 5752df0..90f8f60 100644
--- a/Documentation/devicetree/bindings/usb/usb-xhci.txt
+++ b/Documentation/devicetree/bindings/usb/usb-xhci.txt
@@ -1,14 +1,14 @@
 USB xHCI controllers
 
 Required properties:
-  - compatible: should be "xhci-platform".
+  - compatible: should be "generic-xhci" (deprecated: "xhci-platform").
   - reg: should contain address and length of the standard XHCI
     register set for the device.
   - interrupts: one XHCI interrupt should be described here.
 
 Example:
  usb@f0931000 {
-  compatible = "xhci-platform";
+  compatible = "generic-xhci";
   reg = <0xf0931000 0x8c8>;
   interrupts = <0x0 0x4e 0x0>;
  };
diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 63d0a1b..780cd3e 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -225,6 +225,7 @@ static const struct dev_pm_ops xhci_plat_pm_ops = {
 
 #ifdef CONFIG_OF
 static const struct of_device_id usb_xhci_of_match[] = {
+ { .compatible = "generic-xhci" },
  { .compatible = "xhci-platform" },
  { },
 };
-- 
1.7.1